src/main/java/com/github/minecraftschurlimods/arsmagicalegacy/common/effect/AgilityEffect.java

```diff
@@ -7,24 +7,33 @@
 import net.minecraft.world.entity.LivingEntity;
 import net.minecraft.world.entity.player.Player;
 
+import java.util.HashMap;
+import java.util.Map;
+
 public class AgilityEffect extends AMMobEffect {
+    private static final Map<LivingEntity, Float> CACHE = new HashMap<>();
+
     public AgilityEffect() {
         super(MobEffectCategory.BENEFICIAL, 0xade000);
     }
 
     @Override
     public void startEffect(LivingEntity entity, MobEffectInstance effect) {
-        entity.maxUpStep = 1f;
+        if (entity.maxUpStep >= 1) return;
+        CACHE.put(entity, entity.maxUpStep);
+        entity.maxUpStep = 1;
```

Comment:
this should be replaced with a modifier for the `ForgeMod#STEP_HEIGHT_ADDITION` attribute